integration test: jenkins running the wrong binary but showing the right commit id

while doing this PR: https://github.com/kubernetes/minikube/pull/5530


I noticed a spooky behaviour that my integraiton tests would pass on local machine but fail horribly the jenkins

The logs showed that it had the right commit :

https://storage.googleapis.com/minikube-builds/logs/5530/KVM_Linux.txt
```
Started by upstream project "Build_Cross" build number 8647
originally caused by:
 GitHub pull request #5530 of commit 390992571fbad36aff04fd08fd59e6d5b8a36415, no merge conflicts.
Running as SYSTEM
[EnvInject] - Loading node environment variables.
[EnvInject] - Preparing an environment for the bui
```

but I suspected that it actually runs the right binary !
I added a logging message and I expected it to see in the logs:

```
exit.WithError("Failed to update cluster<--Temproary logging-->", err)
```

however I got this error that doesn't belong to that code !
<details>
<pre>

 X Failed to update cluster: starting kubelet command: "/bin/bash -c \"sudo systemctl daemon-reload && sudo systemctl start kubelet\"" output: "\n** stderr ** \nusage: sudo -h | -K | -k | -V\nusage: sudo -v [-AknS] [-g group] [-h host] [-p prompt] [-u user]\nusage: sudo -l [-AknS] [-g group] [-h host] [-p prompt] [-U user] [-u user]\n            [command]\nusage: sudo [-AbEHknPS] [-C num] [-g group] [-h host] [-p prompt] [-T timeout]\n            [-u user] [VAR=value] [-i|-s] [<command>]\nusage: sudo -e [-AknS] [-C num] [-g group] [-h host] [-p prompt] [-T timeout]\n            [-u user] file ...\n\n** /stderr **": Process exited with status 1
10:42:40         * 
10:42:40         * Sorry that minikube crashed. If this was unexpected, we would love to hear from you:

</pre>
</details>


Also : I see the following failed to delete in the beggining of the script
```
11:03:34 >> Cleaning up after previous test runs ...
11:03:34 * Cleaning stale test root: /home/jenkins/minikube-integration/linux-amd64-kvm2-5225-12310-98e906446c6a6a4dadd13b0a6537dc9f1b3d895a
11:03:34 rmdir: failed to remove '/home/jenkins/minikube-integration/linux-amd64-kvm2-5225-12310-98e906446c6a6a4dadd13b0a6537dc9f1b3d895a': Directory not empty
11:03:34 test.json
11:03:34 test.out
11:03:34 * Cleaning stale test root: /home/jenkins/minikube-integration/linux-amd64-kvm2-5225-29692-c3c62a451c0b089f352485fa9bf668b86bdb5ba4
11:03:34 rmdir: failed to remove '/home/jenkins/minikube-integration/linux-amd64-kvm2-5225-29692-c3c62a451c0b089f352485fa9bf668b86bdb5ba4': Directory not empty
11:03:34 tests.json
11:03:34 * Cleaning stale test root: /home/jenkins/minikube-integration/linux-amd64-kvm2-5225-603-9e58a9a0b93e0f3ba132b98f5ef2df806bf7898b
11:03:34 rmdir: failed to remove '/home/jenkins/minikube-integration/linux-amd64-kvm2-5225-603-9e58a9a0b93e0f3ba132b98f5ef2df806bf7898b': Directory not empty
11:03:34 test.json
11:03:34 * Cleaning stale test root: /home/jenkins/minikube-integration/linux-amd64-none-5225-21635-079c256655b4f84a00b505d8f53b396b52e22885
11:03:34 rmdir: failed to remove '/home/jenkins/minikube-integration/linux-amd64-none-5225-21635-079c256655b4f84a00b505d8f53b396b52e22885': Directory not empty
```

I added the following to the common.sh integration script:
```
echo ">> Starting ${E2E_BIN} at $(date)"
echo ">> The minikube version & commit id:"
echo $(out/minikube-${OS_ARCH} version)
```
and I expected it to see in the logs but !!!! it did not show up, even though the logs point to the right commit ID the common.sh is not the new one ! 

```
11:22:21 >> Starting out/e2e-linux-amd64 at Sat Oct 12 18:22:21 UTC 2019
11:23:13 2019/10/12 18:23:13 [DEBUG] GET http://10.138.0.2:5000
11:25:21 2019/10/12 18:25:21 [DEBUG] GET http://127.0.0.1:40735/api/v1/namespaces/kubernetes-dashboard/services/http:kubernetes-dashboard:/proxy/
11:30:40 --- FAIL: TestStartStop (168.17s)
11:30:40     --- FAIL: TestStartStop/group (168.
```

no sign of the version I added !!!!

--- a/hack/jenkins/minikube_cross_build_and_upload.sh
+++ b/hack/jenkins/minikube_cross_build_and_upload.sh
@@ -63,4 +63,9 @@ cp -r test/integration/testdata out/
 # Don't upload the buildroot artifacts if they exist
 rm -r out/buildroot || true
 
-gsutil -m cp -r out/ "gs://${bucket}/${ghprbPullId}/"
+# At this point, the out directory contains the jenkins scripts (populated by jenkins),
+# testdata, and our build output. Push the changes to GCS so that worker nodes can re-use them.
+
+# -d: delete remote files that don't exist (removed test files, for instance)
+# -J: gzip compression
+gsutil -m rsync -dJ out "gs://${bucket}/${ghprbPullId}"

